securedrop/sdconfig.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+import config as _config
+
+import typing
+# https://www.python.org/dev/peps/pep-0484/#runtime-or-type-checking
+if typing.TYPE_CHECKING:
+    # flake8 can not understand type annotation yet.
+    # That is why all type annotation realted import
+    # statements has to be marked as noqa.
+    # http://flake8.pycqa.org/en/latest/user/error-codes.html?highlight=f401
+    from typing import List, Dict  # noqa: F401
+
+
+class SDConfig(object):
+    def __init__(self):
+        # type: () -> None
+        self.FlaskConfig = _config.FlaskConfig
+        self.JournalistInterfaceFlaskConfig = \
+            _config.JournalistInterfaceFlaskConfig
+        self.SourceInterfaceFlaskConfig = _config.SourceInterfaceFlaskConfig
+
+        if _config.DATABASE_ENGINE != "sqlite":
+            self.DATABASE_USERNAME = _config.DATABASE_USERNAME  # type: ignore
+            self.DATABASE_PASSWORD = \
+                _config.DATABASE_PASSWORD   # type: ignore
+            self.DATABASE_HOST = _config.DATABASE_HOST  # type: ignore
+            self.DATABASE_NAME = _config.DATABASE_NAME  # type: ignore
+        else:
+            self.DATABASE_FILE = _config.DATABASE_FILE
+
+        if hasattr(_config, 'CUSTOM_HEADER_IMAGE'):
+            self.CUSTOM_HEADER_IMAGE = \
+                _config.CUSTOM_HEADER_IMAGE  # type: ignore
+
+        if hasattr(_config, 'SUPPORTED_LOCALES'):
+            self.SUPPORTED_LOCALES = \
+                _config.SUPPORTED_LOCALES  # type: ignore
+
+        # Now we will fill up existing values from config.py
+        if hasattr(_config, "ADJECTIVES"):
+            self.ADJECTIVES = _config.ADJECTIVES  # type: str
+
+        if hasattr(_config, "DATABASE_ENGINE"):
+            self.DATABASE_ENGINE = _config.DATABASE_ENGINE  # type: str
+
+        if hasattr(_config, "DATABASE_FILE"):
+            self.DATABASE_FILE = _config.DATABASE_FILE  # type: str
+
+        if hasattr(_config, "DEFAULT_LOCALE"):
+            self.DEFAULT_LOCALE = _config.DEFAULT_LOCALE  # type: ignore
```

Comment:
I'm curious why we have a few variables that have `# type: ignore`? If this has been answered already and I missed it, we should probably explain in the commit message

Reply:
I will add that in the commit message. They are ignored because just in case the attribute is missing (which we expect) `mypy` will complain. That is why now all the attributes have that. We will improve these in future (after 0.6).